src/Route/RouteCollection.php

```diff
@@ -157,6 +162,16 @@ public function has($name)
         return \array_key_exists($this->getCode($name), $this->elements);
     }
 
+    /**
+     * @param string $name
+     *
+     * @return bool
+     */
+    public function hasCached($name)
```

Comment:
You can and should use php 7 type declarations
```suggestion
    public function hasCached(string $name): bool
```

Reply:
Isn't it weird having some public function final with php 7 type declaration and some without ? How to decide ?

Reply:
> How to decide ?

You can use `final` and type declarations every time that's not a BC-break. So for all new methods, for instance.

Reply:
Ok, I did it for every new function.